sw/device/tests/clkmgr_off_peri_test.c

```diff
@@ -2,41 +2,154 @@
 // Licensed under the Apache License, Version 2.0, see LICENSE for details.
 // SPDX-License-Identifier: Apache-2.0
 
+#include "sw/device/lib/base/abs_mmio.h"
 #include "sw/device/lib/base/memory.h"
 #include "sw/device/lib/dif/dif_aon_timer.h"
 #include "sw/device/lib/dif/dif_base.h"
 #include "sw/device/lib/dif/dif_clkmgr.h"
+#include "sw/device/lib/dif/dif_flash_ctrl.h"
 #include "sw/device/lib/dif/dif_pwrmgr.h"
 #include "sw/device/lib/dif/dif_rstmgr.h"
+#include "sw/device/lib/dif/dif_spi_host.h"
 #include "sw/device/lib/dif/dif_uart.h"
+#include "sw/device/lib/dif/dif_usbdev.h"
 #include "sw/device/lib/runtime/log.h"
 #include "sw/device/lib/testing/aon_timer_testutils.h"
+#include "sw/device/lib/testing/flash_ctrl_testutils.h"
 #include "sw/device/lib/testing/rstmgr_testutils.h"
 #include "sw/device/lib/testing/test_framework/check.h"
 #include "sw/device/lib/testing/test_framework/ottf_main.h"
 
 #include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+#include "spi_host_regs.h"
+#include "uart_regs.h"
+#include "usbdev_regs.h"
 
 /**
  * The peripherals used to test when the peri clocks are disabled are
  * bit 0: clk_io_div4_peri: uart0
  * bit 1: clk_io_div2_peri: spi_host1
- * bit 2: clk_usb_peri: usbdev
- * bit 3: clk_io_peri: spi_host0
+ * bit 2: clk_io_peri: spi_host0
+ * bit 3: clk_usb_peri: usbdev
  */
 
 OTTF_DEFINE_TEST_CONFIG();
 
 static dif_aon_timer_t aon_timer;
+static dif_flash_ctrl_state_t flash_ctrl;
+static dif_spi_host_t spi_host0;
+static dif_spi_host_t spi_host1;
+static dif_usbdev_t usbdev;
 static dif_uart_t uart0;
 
-static const uint32_t bite_us = 200;
+static const uint32_t bite_us = 100;
```

Comment:
@matutem i have a feeling your watchdog timing is just too short here.
The fpga is running at a significantly reduced frequency (for example instead of 24MHz, I think you get 2.5MHz).

So all the utils will still calculate the correct number of cycles for a watchdog, but your actual execution is way slower. If you look at your log you will also see the `set_hung_address` stuff never gets called.

Reply:
Great catch: this used to be 400 us but that was excessive for sim_dv. Fixed.